src/jni_core/src/jvm_runtime.rs

```diff
@@ -160,39 +160,40 @@ pub fn register_native_method_for_jvm(jvm: &JavaVM) -> Result<(), jni::errors::E
 
 /// Load JVM memory statistics from the runtime. If JVM is not initialized or fail to initialize, return zero.
 pub fn load_jvm_memory_stats() -> (usize, usize) {
-    if let Some(jvm) = JVM.get() {
-        match jvm {
-            Ok(jvm) => {
-                let mut env = jvm.attach_current_thread().unwrap();
-                let runtime_instance = env
-                    .call_static_method(
-                        "java/lang/Runtime",
-                        "getRuntime",
-                        "()Ljava/lang/Runtime;",
-                        &[],
-                    )
-                    .unwrap();
-
-                let runtime_instance = match runtime_instance {
-                    JValueOwned::Object(o) => o,
-                    _ => unreachable!(),
-                };
-
-                let total_memory = env
-                    .call_method(runtime_instance.as_ref(), "totalMemory", "()J", &[])
-                    .unwrap()
-                    .j()
-                    .unwrap();
-
-                let free_memory = env
-                    .call_method(runtime_instance, "freeMemory", "()J", &[])
-                    .unwrap()
-                    .j()
-                    .unwrap();
-
-                (total_memory as usize, (total_memory - free_memory) as usize)
+    if let Ok(jvm) = JVM.get() {
+        let result: Result<(usize, usize), jni::errors::Error> = try {
+            let mut env = jvm.attach_current_thread()?;
+
+            let runtime_instance = env.call_static_method(
```

Comment:
Previously, we used OnceLock here to avoid initializing a JVM when the memory controller collects jvm memory statistics.
https://github.com/risingwavelabs/risingwave/pull/12965#discussion_r1366511055

Reply:
Good catch! I have changed to avoid initialize the jvm when collecting metrics.